guids-generator.html

```diff
@@ -462,21 +462,23 @@
                             return;
                         }
 
-                        // Remove salt version, rounds number and actual salt from hash
-                        // Encode the hash in HEX to ensure only alphanumeric characters are generated
-                        hash = hash.replace(salt, '')
-                                 .split("")
-                                 .map((c, i) => {
-                                                   let str = c.charCodeAt(0).toString(16).padStart(2, "0");
-                                                   // Add random noise in alternating bytes
-                                                   str += Math.random().toString(16).substring(2, 4);
-                                                   return str;
-                                                })
-                                 .join("");
+                        /*
+                         Turn the base64 encoded hash (remove salt version, rounds number and actual salt)
+                         into a Typed Array that we can work with byte-by-byte.
+                        */
+                        hash = Uint8Array.from(window.atob(hash.slice(-31)), (c) => c.charCodeAt(0));
+
+                        // Interlace the hash with random noise bytes
+                        let noisyhash = new Uint8Array(2 * hash.length);
+                        for (let i = 0; i < hash.length; i++) {
```

Comment:
Wasn't the proposal to split this into blocks of 16 bits (2 bytes), so 2 bytes hash, 2 bytes noise? I don't think the security changes either way, just wanted to make sure this is a conscious decision to not use 16b interlacing.

Reply:
I implemented this based on the specifications on _page 6_ of the proposal. _eGUID generation (II)_ states _a byte of GUID followed by a byte of noise..._ and _Figure 2_ shows the noise in red as groups of two hexadecimal digits (that is, one byte).

Reply:
Ah, indeed, it was two chars, not two bytes.